resources/database.rb

```diff
@@ -0,0 +1,83 @@
+#
+# Cookbook:: postgresql
+# Resource:: user
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+resource_name :postgresql_database
+
+property :name,     kind_of: String, name_attribute: true
```

Comment:
I think the best thing here would be to change this to db_name which would allow people to override the resource name and set the name as a property. You'd need to go through and replace all the new_resource.name occurrences with new_resource.db_name.